url80587fc7d5_dennervidal_CloudSim_DVFS_tgz-pJ8-Examples_Dvfs_example_simpleJ8/src/org/cloudbus/cloudsim/workflow/Configuration.java

```diff
@@ -0,0 +1,39 @@
+package org.cloudbus.cloudsim.workflow;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Properties;
+
+/**
+ * This class implements the properties manager. Here the
+ * name of the properties file is defined. Moreover, when
+ * properties are to be read, this is the class that is called.
+ *
+ */
+public enum Configuration {
+	
+	INSTANCE;
+		
+	private static Properties properties = System.getProperties();
+
+	static {
+		loadPropertiesFromFile("simulation.properties");
+	}
+		 
+    public String getProperty(String key) {
+        return properties.getProperty(key);
+    }
+    
+    public void setProperty(String key, String value) {
+        properties.setProperty(key, value);
+    }
+    
+    private static void loadPropertiesFromFile(String file) {
+        File propertiesFile = new File(file);
+        try {
+            properties.load(new FileInputStream(propertiesFile));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor: *FileInputStream*. Execution paths that do not contain closure statements were detected. Ensure the object returned by *FileInputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).